zerostash/src/commands/mount.rs

```diff
@@ -0,0 +1,279 @@
+        zerostash_files::FileType::Symlink(_) => FileType::Symlink,
+        zerostash_files::FileType::Directory => panic!("Didnt expect a directory!"),
+    }
+}
+
+impl SimpleFs {
+    pub fn open(
+        stash: Infinitree<Files>,
+        options: &restore::Options,
+        destroy_tx: mpsc::SyncSender<()>,
+    ) -> Result<Self> {
+        let mut dir_map: HashMap<DirPath, Vec<DirectoryEntry>> = HashMap::new();
+        dir_map.insert(PathBuf::new(), vec![]);
+        let mut file_map = HashMap::new();
+
+        for file in options.list(&stash) {
+            let file_name_path = PathBuf::from(&file.name);
+            let filetype = match_filetype(&file.file_type);
+
+            add_dir_to_map(&mut dir_map, &file_name_path, filetype);
+            file_map.insert(file_name_path, file.clone());
+        }
+
+        Ok(SimpleFs {
+            destroy_tx,
+            stash,
+            file_map,
+            file_content: Mutex::new(HashMap::new()),
+            dir_map,
+            file_attr_memory: Mutex::new(HashMap::new()),
+        })
+    }
+}
+
+const TTL: Duration = Duration::from_secs(1);
+
+pub fn file_to_fuse(file: &Arc<Entry>) -> FileAttr {
+    FileAttr {
+        size: file.size,
+        blocks: 1,
+        atime: SystemTime::now(),
+        mtime: SystemTime::now(),
+        ctime: SystemTime::now(),
+        crtime: SystemTime::UNIX_EPOCH,
+        kind: FileType::RegularFile,
+        perm: 0o444,
+        nlink: 1,
+        gid: file.unix_gid.unwrap_or(0),
+        uid: file.unix_uid.unwrap_or(0),
+        rdev: 0,
+        flags: 0,
+    }
+}
+
+fn strip_path(path: &Path) -> &Path {
+    path.strip_prefix("/").unwrap()
+}
+
+impl FilesystemMT for SimpleFs {
+    fn destroy(&self) {
+        self.destroy_tx
+            .send(())
+            .expect("Could not send signal on channel.")
+    }
+    fn getattr(&self, _req: RequestInfo, path: &Path, _fh: Option<u64>) -> ResultEntry {
+        println!("getattr = {:?}", path);
+
+        let real_path = strip_path(path);
+        let mut file_attr_memory = self.file_attr_memory.lock().unwrap();
+
+        if self.dir_map.contains_key(real_path) {
+            Ok((TTL, DIR_ATTR))
+        } else if let Some(file_attr) = file_attr_memory.get(real_path) {
+            Ok((TTL, *file_attr))
+        } else {
+            match self.file_map.get(real_path) {
+                Some(metadata) => {
+                    let fuse = file_to_fuse(metadata);
+                    file_attr_memory.insert(real_path.into(), fuse);
+                    Ok((TTL, fuse))
+                }
+                None => Err(libc::ENOENT),
+            }
+        }
+    }
+
+    fn opendir(&self, _req: RequestInfo, _path: &Path, _flags: u32) -> ResultOpen {
+        println!("(opendir: {:?} flags = {:#o})", _path, _flags);
+        Ok((0, 0))
+    }
+
+    fn readdir(&self, _req: RequestInfo, path: &Path, _fh: u64) -> ResultReaddir {
+        println!("readdir: {:?}", path);
```

Comment:
Use the `tracing` crate instead of `println`s, and these should be `debug` severity messages.